### ISSUE TITLE
BUG: SSM does not support secretsmanager references

### ISSUE THREAD

**Issue by abbottdev** [localstack/localstack#3128]
# Type of request: Bug Report

Filed as bug because it localstack does not emulate AWS SSM Service's behaviour

# Detailed description

The AWS documentation states that the SSM service should support AWS Secret references (source: https://docs.aws.amazon.com/systems-manager/latest/userguide/integration-ps-secretsmanager.html)

When you try to resolve a secret in this way via Localstack you receive an error stating the parameter doesn't exist - but SSM should act as a pass-through to secrets manager using a special reserved parameter prefix.

...

## Expected behavior
Running `awslocal ssm get-parameter --name /aws/reference/secretsmanager/dbSecret` returns data like so:
```
{
    "Parameter": {
        "Name": "/aws/reference/secretsmanager/dbSecret",
        "Value": "{\"password\": \"MY_SECRET_PWD\"}",
        "Type": "SecureString",
        "LastModifiedDate": 2020-09-28T21:47:14.743Z,
        "ARN": "arn:aws:secretsmanager:eu-west-2:1234567890:secret:dbSecret-vohxj",
        "SourceResult": 
              "{
                   \"CreatedDate\": 1526334434.743,
                   \"Name\": \"dbSecret\",
                   \"VersionId\": \"aaabbbccc-1111-222-333-123456789\",
                   \"SecretString\": \"{\\"password\\": \\"MY_SECRET_PWD\\"}\",
                   \"VersionStages\": [\"AWSCURRENT\"],
                   \"ARN\": \"arn:aws:secretsmanager:eu-west-2:1234567890:secret:dbSecret-vohxj\"
               }"
      }
```

...

## Actual behavior
Error: `An error occurred (ParameterNotFound) when calling the GetParameter operation: Parameter /aws/reference/secretsmanager/dbSecret not found.`
...

# Steps to reproduce

`awslocal secretsmanager create-secret --name dbSecret --secret-string "{\"password\": \"MY_SECRET_PWD\"}"`

`awslocal secretsmanager get-secret-value --secret-id dbSecret`

```
{
    "ARN": "arn:aws:secretsmanager:eu-west-2:1234567890:secret:dbSecret-vohxj",
    "Name": "dbSecret",
    "VersionId": "72340ebd-8e56-455b-8d0f-51683e909076",
    "SecretString": "{\"password\": \"MY_SECRET_PWD\"}",
    "VersionStages": [
        "AWSCURRENT"
    ],
    "CreatedDate": 1603102652
}
```

`awslocal ssm get-parameter --name /aws/reference/secretsmanager/dbSecret`

`An error occurred (ParameterNotFound) when calling the GetParameter operation: Parameter /aws/reference/secretsmanager/dbSecret not found.`

This error should not occur and should instead follow the AWS documentation and resolve the secret value.

## Command used to start LocalStack

Docker compose w/ localstack/localstack image.
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by abbottdev** [comment]
Also worthwhile noting - you can't workaround this issue by issuing a `put parameter` request because the AWS-CLI blocks it:
`An error occurred (AccessDeniedException) when calling the PutParameter operation: No access to reserved parameter name: xxx`

**Comment by whummer** [comment]
Thanks for reporting @abbottdev . This should be fixed in #3171 - can you please pull the `latest` Docker image and give it another try..? Please report here if the problem persists. Thanks
